spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/TraceWebFilter.java

```diff
@@ -156,6 +158,11 @@ TraceKeys traceKeys() {
 								addClassMethodTag(handlerMethod, span);
 								addClassNameTag(handlerMethod, span);
 							}
+							Object httpRoute = exchange
+									.getAttribute(HandlerMapping.BEST_MATCHING_PATTERN_ATTRIBUTE);
```

Comment:
the contract is empty string when route is available (which I think is always the case here), just no match.

Ex. in webmvc we do..
```
    Object httpRoute = request.getAttribute(BEST_MATCHING_PATTERN_ATTRIBUTE);
    request.setAttribute("http.route", httpRoute != null ? httpRoute.toString() : "");
```

So here probably

```
    Object httpRoute = request.getAttribute(BEST_MATCHING_PATTERN_ATTRIBUTE);
    response.httpRoute = httpRoute != null ? httpRoute.toString() : "";
```

Reply:
There you already know for sure that request is handled by MVC. In this case it is not always true (think reactive functions for example). Also I was not able to reproduce null `BEST_MATCHING_PATTERN_ATTRIBUTE` while staying in MVC context. However I noticed following corner cases:
- default error page provides `/**` as a pattern
- for catch-all `GetMapping` provided pattern equals to URI - whatever the client supplies